test/functional/toolchains/test_premake.py

```diff
@@ -1,74 +1,124 @@
+import os
 import platform
 import textwrap
 
 import pytest
 
-from conan.test.assets.sources import gen_function_cpp
+from conan.test.utils.tools import TestClient
 
 
 @pytest.mark.skipif(platform.system() != "Linux", reason="Only for Linux now")
 @pytest.mark.tool("premake")
-def test_premake(matrix_client):
```

Comment:
Better not change existing tests, unless they are considered broken.
Because these tests protects against breaking existing users.

Reply:
Okey, restored old test and keep old behaviour when `PremakeToolchain` is not used